### ISSUE TITLE
Feature request: winhttp HTTP back-end on windows

### ISSUE THREAD

**Issue by ethindp** [stcarrez/ada-util#54]
@stcarrez Would it be possible to have a winhttp back-end for HTTP in utilada? According to version 2.7.0, AWS 24 is broken on Windows, and though cURL sort-of works, it doesn't use the Windows certificate store for TLS/SSL certificates. This is definitely a big, big problem and I would even go so far as to say it's a bad thing because that is what people will expect it to do. This however is not to my knowledge something that ada-util can control, other than possibly implementing an alternative back-end like this.

**Comment by stcarrez** [comment]
@ethindp, I agree this is unfortunate but I cannot control that.  It is reported in https://github.com/alire-project/alire/issues/1710 from the first crate submission I've made for Ada Util 2.7.0 (GitHub Windows action failed) and from a user on ada-lang.io forum.

I can try to enable it again and try to submit an updated crate with Windows enabled as before and we will see if it works. I suspect that the way @Fabien-Chouteau build it recently on his Windows machine is different than the GitHub automated build which failed.
